datastream/client.go

```diff
@@ -201,6 +203,42 @@ func (c *DataStreamClient) handleFlagsMessage(ctx context.Context, resp *schemat
 	return nil
 }
 
+func (c *DataStreamClient) handleFlagMessage(ctx context.Context, resp *schematicdatastreamws.DataStreamResp) error {
+	flags := resp.Data
+
+	var flag *rulesengine.Flag
+	if err := json.Unmarshal(flags, &flag); err != nil {
+		return fmt.Errorf("Failed to unmarshal flags data: %v", err)
+	}
+
+	c.flagsMu.Lock()
+	defer func() {
+		c.flagsMu.Unlock()
+
+		if c.pendingFlagRequest != nil {
+			c.pendingFlagRequest <- true
+		}
+	}()
+
+	cacheKey := flagCacheKey(flag.Key)
+	switch resp.MessageType {
+	case schematicdatastreamws.MessageTypeDelete:
```

Comment:
The server doesn't actually send delete message types yet, but I could see us adding it in the future; seems good to just implement now so that support is ready